src/Data/StringTable.hs

```diff
@@ -42,7 +42,7 @@ instance MemSize (StringTable id) where
 lookup :: Enum id => StringTable id -> String -> Maybe id
 lookup (StringTable bs tbl) str = binarySearch 0 (topBound-1) (BS.pack str)
   where
-    (0, topBound) = A.bounds tbl
+    (_assumedZero, topBound) = A.bounds tbl
```

Comment:
I don't think this is an improvement, since the algorithm now runs further if the `_assumedZero` isn't actually zero. But should it run further? It's basically the question of whether you want `assert`s in prod or not.

Personally, I think partiality is better than an incorrect result. Even though the compiler warns for partiality, but it doesn't warn for incorrectness. Of course, it would be even better to prove the algorithm correct. But that isn't always feasible, and the lack of dependent types make it extra difficult.